internal/client/graphql.go

```diff
@@ -0,0 +1,46 @@
+package client
+
+import (
+	"errors"
+	"strings"
+)
+
+type graphQLRequest struct {
+	operationName string
+	query         string
+	variables     map[string]interface{}
+}
+
+func (g *graphQLRequest) build() map[string]interface{} {
+	return map[string]interface{}{
+		"operationName": g.operationName,
+		"query":         g.query,
+		"variables":     g.variables,
+	}
+}
+
+type Error struct {
+	Message string   `json:"message"`
+	Path    []string `json:"path"`
+}
+
+func (e *Error) Error() string {
+	return e.Message + " " + strings.Join(e.Path, ".")
+}
+
+func coalesceErrors[T any](resp Response[T], err error) error {
```

Comment:
cool, to generalize? :)

Reply:
the generic here is required because of how the struct is defined